met-api/src/met_api/resources/language.py

```diff
@@ -75,43 +76,82 @@ def delete(language_id):
 
     @staticmethod
     @cross_origin(origins=allowedorigins())
-    def get():
-        """Fetch list of languages."""
+    def get(tenant_id):
+        """Fetch list of languages associated with a given tenant ID."""
         try:
-            languages = LanguageService.get_languages()
+            languages = LanguageService.get_languages_by_tenant(tenant_id)
             return (
                 jsonify(LanguageSchema(many=True).dump(languages)),
                 HTTPStatus.OK,
             )
         except (KeyError, ValueError) as err:
             return str(err), HTTPStatus.INTERNAL_SERVER_ERROR
 
+
+@cors_preflight('POST, OPTIONS')
+@API.route('/<int:language_id>/tenant/<int:tenant_id>')
+class ManageLanguageMappings(Resource):
+    """Resource for adding or removing language-tenant relationships."""
+
+    @staticmethod
+    def post(language_id, tenant_id):
+        """Create language-tenant mapping."""
+        try:
+            return LanguageService.map_language_to_tenant(language_id, tenant_id)
+        except IntegrityError as e:
+            # Catching language code already exists error
+            detail = (
+                str(e.orig).split('DETAIL: ')[1]
+                if 'DETAIL: ' in str(e.orig)
+                else 'Duplicate entry.'
+            )
+            raise BusinessException(
+                str(detail), HTTPStatus.INTERNAL_SERVER_ERROR
+            ) from e
+
+
+@cors_preflight('DELETE, OPTIONS')
+@API.route('/mappings/<int:language_mapping_id>')
+class DeleteLanguageMapping(Resource):
+    """Resourse for removing language-tenant relationships."""
+
+    @staticmethod
+    # @_jwt.requires_auth
+    # @cross_origin(origins=allowedorigins())
+    def delete(language_mapping_id):
+        """Remove a language mapping from a tenant."""
+        try:
+            success = LanguageService.remove_language_mapping_from_tenant(language_mapping_id)
+            if success:
+                return 'Successfully deleted language-tenant mapping', HTTPStatus.NO_CONTENT
+            raise ValueError('Language-tenant mapping not found')
+        except KeyError as err:
+            return str(err), HTTPStatus.BAD_REQUEST
+        except ValueError as err:
+            return str(err), HTTPStatus.NOT_FOUND
+
+
+@cors_preflight('GET, POST, OPTIONS, PATCH, DELETE')
+@API.route('/')
+class Languages(Resource):
+    """Resource for managing multiple languages."""
+
     @staticmethod
-    @_jwt.requires_auth
     @cross_origin(origins=allowedorigins())
-    def post():
-        """Create a new language."""
+    def get():
+        """Fetch list of languages."""
         try:
-            request_json = request.get_json()
-            valid_format, errors = schema_utils.validate(
-                request_json, 'language'
+            languages = LanguageService.get_languages()
+            return (
+                jsonify(LanguageSchema(many=True).dump(languages)),
```

Comment:
Is this call to `jsonify` necessary if dump() already produces a json string?

Reply:
Oh I missed this, this might just solve an issue I was having... Thanks